libs/SalesforceSDKCore/SalesforceSDKCore/Classes/OAuth/SFOAuthCoordinator.m

```diff
@@ -935,20 +936,20 @@ + (NSString *)advancedAuthStateDesc:(SFOAuthAdvancedAuthState)authState
 
 - (NSURLSession*)session {
     if (_session == nil) {
-        _session = [NSURLSession sessionWithConfiguration:[NSURLSessionConfiguration ephemeralSessionConfiguration]];
+        SFNetwork *network = [[SFNetwork alloc] init];
+        _session = network.activeSession;
```

Comment:
The value here is to consolidate all requests that go through the `Mobile SDK` through a single network stack. This ensures that all requests have the same `NSURLConfiguration`, which is important in some corner cases. I recently discovered that there's a server side configuration that allows the server to reject requests that originate from a different IP subnet. This would fail if `TwinPrime` configuration was configured in `SFNetwork` but these requests used their own `NSURLConfiguration`, because the IP addresses would be different and the server would reject the valid `Bearer` token.